### ISSUE TITLE
Accessibility: Focusable div at beginning of modals

### ISSUE THREAD

**Issue by Watso196** [ant-design/ant-design#48903]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/antd-reproduction-template-forked-sq6sg2?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clvwnwz4p00063b6jcabnrfmi%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clvwnwz4p00023b6jmejfebl7%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clvwnwz4p00033b6j2npby06m%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clvwnwz4p00053b6jf7x71z1h%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clvwnwz4p00023b6jmejfebl7%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvwnwz4p00013b6ji14yjsiq%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A10%252C%2522startColumn%2522%253A39%252C%2522endLineNumber%2522%253A10%252C%2522endColumn%2522%253A39%257D%255D%257D%255D%252C%2522id%2522%253A%2522clvwnwz4p00023b6jmejfebl7%2522%252C%2522activeTabId%2522%253A%2522clvwnwz4p00013b6ji14yjsiq%2522%257D%252C%2522clvwnwz4p00053b6jf7x71z1h%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvwnwz4p00043b6jqthljola%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clvwnwz4p00053b6jf7x71z1h%2522%252C%2522activeTabId%2522%253A%2522clvwnwz4p00043b6jqthljola%2522%257D%252C%2522clvwnwz4p00033b6j2npby06m%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clvwnwz4p00033b6j2npby06m%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Steps to reproduce

1. Open modal
2. Use keyboard Tab key to move focus
3. Note that focus is "lost" (become invisible) after moving past the OK button, due to focus moving to the hidden div at the top of the modal

### What is expected?

Focus should stay within the modal, moving between interactive elements, and should not disappear at any point

### What is actually happening?

Focus visually disappears when landing on the non-interactive and invisible hidden <div> at the top of the modal

| Environment | Info |
| --- | --- |
| antd | 5.17.0 |
| React | 17 |
| System | Mac Sonoma 14.4.1 (23E224) |
| Browser | Version 124.0.6367.158 (Official Build) (arm64) |

---
This is technically a failure of the Focus Visible and Focus Order WCAG guidelines:
Focus Visible: https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html
Focus Order: https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html

**Comment by scottapow** [comment]
Looks like there was an [attempted resolution for this issue here](http://local.apci.org:3001) but it still persists for me after 5.15.0. You might consider moving this issue to the [rc-dialog repo](https://github.com/react-component/dialog) @Watso196

**Comment by Watso196** [comment]
Gotcha, thanks @scottapow! I've submitted this issue to the rc-dialog repo
